src/api/layer.js

```diff
@@ -331,22 +332,20 @@ export default class Layer {
      * @param {Dataframe} dataframe
      */
     _onDataframeAdded(dataframe) {
+        dataframe.freeObserver = () => {
+            this._renderLayer.removeDataframe(dataframe);
```

Comment:
This function [is removed in this PR](https://github.com/CartoDB/carto-vl/pull/487/files#diff-f276b6fefb280b9dbe9c2b44aa3c035eL25) I think its gonna crash